In [1]:
import pandas as pd
import numpy as np

In [3]:
raw_data_path = "data/raw/WA_Fn-UseC_-Telco-Customer-Churn.csv"
print("Sucessfully path loaded")

Sucessfully path loaded


In [7]:
df = pd.read_csv(raw_data_path)
print("data loaded successfully")

data loaded successfully


In [8]:
df.head()

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   

               PaymentMethod MonthlyCharges  TotalCharges Churn  
0           Electronic check          29.85         29.85    No  
1               Mailed check          56.95        1889.5    No  
2               Mailed check          53.85        108.15   Yes  
3  Bank transfer (automatic)          42.30       1840.75    No  
4           Electronic check          70.70        151.65   Yes  

[5 rows x 21 columns]

In [9]:
pd.set_option('display.max_columns',None)
df.head(5)

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity OnlineBackup  \
0  No phone service             DSL             No          Yes   
1                No             DSL            Yes           No   
2                No             DSL            Yes          Yes   
3  No phone service             DSL            Yes           No   
4                No     Fiber optic             No           No   

  DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0               No          No          No              No  Month-to-month   
1              Yes          No          No              No        One year   
2               No          No          No              No  Month-to-month   
3              Yes         Yes          No              No        One year   
4               No          No          No              No  Month-to-month   

  PaperlessBilling              PaymentMethod  MonthlyCharges TotalCharges  \
0              Yes           Electronic check           29.85        29.85   
1               No               Mailed check           56.95       1889.5   
2              Yes               Mailed check           53.85       108.15   
3               No  Bank transfer (automatic)           42.30      1840.75   
4              Yes           Electronic check           70.70       151.65   

  Churn  
0    No  
1    No  
2   Yes  
3    No  
4   Yes

In [14]:
df.shape

(7043, 21)

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


In [16]:
df.describe()

SeniorCitizen       tenure  MonthlyCharges
count    7043.000000  7043.000000     7043.000000
mean        0.162147    32.371149       64.761692
std         0.368612    24.559481       30.090047
min         0.000000     0.000000       18.250000
25%         0.000000     9.000000       35.500000
50%         0.000000    29.000000       70.350000
75%         0.000000    55.000000       89.850000
max         1.000000    72.000000      118.750000

In [17]:
df.columns

Index(['customerID', 'gender', 'SeniorCitizen', 'Partner', 'Dependents',
       'tenure', 'PhoneService', 'MultipleLines', 'InternetService',
       'OnlineSecurity', 'OnlineBackup', 'DeviceProtection', 'TechSupport',
       'StreamingTV', 'StreamingMovies', 'Contract', 'PaperlessBilling',
       'PaymentMethod', 'MonthlyCharges', 'TotalCharges', 'Churn'],
      dtype='object')

Column: customerID  
Business meaning: unique serial number given to every customer
Type: object  
Potential churn influence: No

Column: gender  
Business meaning: Sex of the customer
Type: object  
Potential churn influence: Yes

Column: SeniorCitizen  
Business meaning: Categorized customers according to there citized ship
Type: object 
Potential churn influence: Yes

Column: Partner  
Business meaning: does customer holds any partnership with company
Type: object  
Potential churn influence: yes

Column: Dependents  
Business meaning: are customers depanding on company or not
Type: object  
Potential churn influence: yes

Column: tenure
Business meaning: The number of months or years the customer stayed with company
Type: int64  
Potential churn influence: Yes

Column: PhoneService
Business meaning: The customer has any phone services with the company
Type: object  
Potential churn influence: Yes

Column: MultipleLines
Business meaning: The customer have more than 1 phone services with the company
Type: object  
Potential churn influence: yes

Column: InternetService  
Business meaning: The customer has enabled internet service/which type of internet sevice from the company
Type: object  
Potential churn influence: yes

Column: OnlineSecurity 
Business meaning: Customer has enabled security service through online 
Type: object  
Potential churn influence: yes

Column: OnlineBackup  
Business meaning: customer has enabled backup storage through online
Type: object  
Potential churn influence: yes

Column: DeviceProtection  
Business meaning: customer activated protection for the device or not
Type: object  
Potential churn influence: yes

Column: TechSupport  
Business meaning: customer enabled for technical support and updates
Type: object  
Potential churn influence: yes

Column: StreamingTV  
Business meaning: the customer has enabled online tv services or not
Type: object  
Potential churn influence: yes

Column: StreamingMovies 
Business meaning: the customer has enabled online movie services or not
Type: object  
Potential churn influence: No

Column: Contract
Business meaning: the customer has contract based or full time based
Type: object  
Potential churn influence: yes

Column: PaperlessBilling
Business meaning: the customer has enabled paper less billing or not
Type: object  
Potential churn influence: yes

Column: PaymentMethod  
Business meaning: what type of payment customer choose
Type: object  
Potential churn influence: yes

Column: MonthlyCharges
Business meaning: unique serial number given to every customer
Type: float64 
Potential churn influence: yes

Column: TotalCharges
Business meaning: What are the annual charged applied for the customer
Type: object  
Potential churn influence: yes

Column: Churn
Business meaning: Does the customer still holds the relationship with the company
Type: object 
Potential churn influence: yes


# Key Take Aways

- The Total dataset consists of 7043 rows and 21 columns
- There columns consists of 
    - 18 objects
    - 2 ints
    - 1 float
    - *issue* is colume "Total charge" is an float dtype but it is recoreded as object
- Key columns for *churn* are: gender, partner, dependents, tenure, phoneservice, online backup, device protection, tech support, streaming tv, contract, paperless billing, payment method, monthly charges, total charges, churn.